core/src/test/java/jenkins/util/io/PathRemoverTest.java

```diff
@@ -418,6 +423,34 @@ public void testForceRemoveRecursive_ParentIsSymbolicLink() throws IOException {
         assertFalse(d1.exists());
     }
 
+    @Test
+    @Issue("JENKINS-55448")
+    public void testForceRemoveRecursive_AbortsAfterTooManyErrors() throws IOException {
```

Comment:
I just added one basic test for now, but if anyone sees an edge case that is not covered by this test or has a specific case they want to test, let me know and I can work on adding more tests.